Automattic/wp-calypso
# Backup: improve first backup UI

Resolves https://github.com/Automattic/jetpack-backup-team/issues/425

The purpose of this PR is to address some feedback received in p6TEKc-7Ef-p2 to improve the initial backup experience by removing or disabling unnecessary UI elements, and to start monitoring how many users are seeing this view and clicking on the contact support link.

## Proposed Changes
* Add `siteHasBackups` selector, decoupling it from the date picker component.
* Hide `Copy site` if site has no backups yet.
* Update `Back up now` and `Copy site` buttons to use `@automattic/components` so it can use the same border-radius and color than other buttons in Calypso.
* Disable date picker if site has no backups yet.
* Remove `Visit your website` and `Manage your website` links.
* Update status cards paragraphs are no longer restricted to 500px on larger screens.
* Add `calypso_jetpack_backup_first_time_view` track event when the `no backups yet` view is rendered
* Add `calypso_jetpack_backup_first_time_support_click` track event when user clicks on the `Contact support` in `no backups yet` view.

| Before | After |
|---|---|
| ![CleanShot 2024-03-29 at 15 15 56@2x](https://github.com/Automattic/wp-calypso/assets/1488641/771187bd-e409-4926-82e5-86d2fe9dfcb4) | ![CleanShot 2024-03-29 at 15 04 22@2x](https://github.com/Automattic/wp-calypso/assets/1488641/fc6f6679-3dc8-4cda-935b-b53148477cdb) |

## Testing Instructions


* Create a new Jurassic Ninja site with Jetpack VaultPress Backup plan and connect it to Jetpack.
* Spin up a Jetpack Cloud live branch
* Pick your new site
* Navigate to the backup page
* Ensure you see all changes described above. Currently, the initial backup is not being triggered automatically for Jurassic Ninja sites, so this screen is easy to reproduce for now.
* Queue a full backup (you can use "Back up now" button
* When it is completed, refresh the page and ensure you see the `Copy site` button and the datepicker is now enabled.
  ![CleanShot 2024-03-31 at 23 15 31@2x](https://github.com/Automattic/wp-calypso/assets/1488641/62402e75-3400-4a48-8eb4-3d87e7967268)

### Validating Tracks events
* Open your developer console in Network tab
* When you access the backup page where the `Your first backup will be available soon` you should see the `t.gif` triggering the `calypso_jetpack_backup_first_time_view`event.
* Click on `Contact support` and you should see the `t.gif` triggering the `calypso_jetpack_backup_first_time_support_click` event.

## Pre-merge Checklist
